damnit/backend/db_migrations.py

```diff
@@ -0,0 +1,181 @@
+import shutil
+from dataclasses import dataclass
+from datetime import datetime, timezone
+from pathlib import Path
+from textwrap import dedent
+from typing import Callable, Iterable
+
+import sqlite3
+
+
+@dataclass(frozen=True)
+class Migration:
+    """
+    A single schema migration step applied to go to `to_version`.
+
+    - `to_version`: schema version after this migration completes
+    - `description`: short human description for docs/logs
+    - `apply`: a callable that takes a sqlite3.Connection and applies changes
+    """
+
+    to_version: int
+    description: str
+    apply: Callable[[sqlite3.Connection], None]
+
+
+def _to_v1(conn: sqlite3.Connection) -> None:
+    conn.executescript(
+        dedent(
+            """
+            -- Core schema (baseline)
```

Comment:
Further stylistic comment: a lot of this file ends up 12 spaces indented. I think it's a good idea to keep the flexibility of a callable, not just an SQL script, but I wonder if it's worth a shortcut wrapping a string for the cases where it is just SQL?

Alternatively, maybe we just reformat it to knock off some whitespace:


```suggestion
    conn.executescript(dedent(
        """
        -- Core schema (baseline)
```